### ISSUE TITLE
[Gitlab] Switch to turn off failing of pipeline

### ISSUE THREAD

**Issue by bmaehr** [mc1arke/sonarqube-community-branch-plugin#137]
**Describe the Enhancement**
Switch to turn off failing of pipeline

**Additional context**
If quality gate is not reached it is not possible to merge, which is to hard for us

**Comment by matthiasbalke** [comment]
You could reconfigure your quality gate, that it only fails if your code get's worse than it was before. But keeping it as "bad" as it was in the target branch would not fail the gate.

Would that be an option?

**Comment by bmaehr** [comment]
No, that's not a option. 

For example if you have a MR with only configuration changes, then the test coverage is zero. So you would have to change the code coverage for new code to 0% at the quality gate to accept the the MR.
Or an other case: Sometime it happens that it is necessary to merge a MR even if the quality gate fails. And usually such things happen when you are under extreme time pressure.

'Must' conditions in software development sounds good until you face the real world. That's why usually all quality solutions have the possibility to allow exceptions (@Ignore at tests, //NOSONAR, /* NOFORMAT , mark as false positive in Sonar, ...). And I think the approval of the MR is a good person to decide if a MR should be merged even if the quality gate fails.

**Comment by bmaehr** [comment]
I have reimplemented the change based on master if someone is interested ...

**Comment by szpak** [comment]
I also face a problem that with enabled quality gate, for the interim period, I would like to have the external job created from SonarQube to do not fail the pipeline in MRs and it should be controlled from the job that triggers Sonar (e.g. with `allow_failure: true`). Is there currently any built-in way to achieve that (even by disabling a pipeline decoration with that "external jobs")?

@bmaehr 1. Am I right that your changes in https://github.com/bmaehr/sonarqube-community-branch-plugin/commits/canFailPipeline cover that?
2. Have you created a PR to have it included in the upstream version (I wasn't able to find any)?

**Comment by szpak** [comment]
@mc1arke Is there currently any way to disable failing the whole pipeline in GitLab (e.g. when waiting for quality gate is done within the job with `allow_failure: true`) or even disable "changing the GitLab pipeline status" (`submitPipelineStatus`) at all?

If not, do you have any better idea how to achieve that?

**Comment by mc1arke** [comment]
@szpak No, there isn't a way within the plugin. You're free to deploy your own copy of the plugin with alterations to the submission handling though.

**Comment by szpak** [comment]
Thanks for your reply @mc1arke . Assuming I would be done in my "copy" of the plugin (a possibility to disable submitting the pipeline status update), were you open to accept that PR?

**Comment by szpak** [comment]
@mc1arke I've created a PR: #637.
